src/pip/_internal/operations/prepare.py

```diff
@@ -329,19 +329,62 @@ def _get_linked_req_hashes(self, req: InstallRequirement) -> Hashes:
         # showing the user what the hash should be.
         return req.hashes(trust_internet=False) or MissingHashes()
 
+    def _fetch_metadata_only(
+        self,
+        req: InstallRequirement,
+    ) -> Optional[BaseDistribution]:
+        if self.require_hashes:
+            logger.debug(
+                "Metadata-only fetching is not used as hash checking is required",
+            )
+            return None
+        # Try PEP 658 metadata first, then fall back to lazy wheel if unavailable.
+        return self._fetch_metadata_using_pep_658(
+            req
+        ) or self._fetch_metadata_using_lazy_wheel(req.link)
+
+    def _fetch_metadata_using_pep_658(
```

Comment:
```suggestion
    def _fetch_metadata_using_link_data_attr(
```

I'd prefer to avoid having PEP numbers in implementation (CPython doesn't implement f-string as `PyString_PEP498` :P).

Reply:
This sounds great! Will keep this in mind for the future :D